src/helpers/requestIdleCallbackAsync.ts

```diff
@@ -0,0 +1,16 @@
+import type { UnboxPromise } from '@/types/index.js';
+
+export function requestIdleCallbackAsync<T extends () => Promise<unknown>, R = UnboxPromise<ReturnType<T>>>(
```

Comment:
There is a builtin generic utility `Awaited`
```suggestion
export function requestIdleCallbackAsync<T extends () => Promise<unknown>, R = Awaited<ReturnType<T>>>(
```